core/drivers/clk/clk.c

```diff
@@ -7,19 +7,38 @@
 #include <config.h>
 #include <drivers/clk.h>
 #include <kernel/boot.h>
+#include <kernel/mutex_pm_aware.h>
 #include <kernel/panic.h>
-#include <kernel/spinlock.h>
+#include <kernel/thread.h>
 #include <malloc.h>
 #include <stddef.h>
 #include <stdio.h>
 
-/* Global clock tree lock */
-static unsigned int clk_lock = SPINLOCK_UNLOCK;
+/* Global clock tree lock/mutex */
+static struct mutex_pm_aware mu = MUTEX_PM_AWARE_INITIALIZER;
 
 #ifdef CFG_DRIVERS_CLK_PRINT_TREE
 static SLIST_HEAD(, clk) clock_list = SLIST_HEAD_INITIALIZER(clock_list);
 #endif
 
+/*
+ * Clock operation may occur at runtime and during specific
+ * system power transition (PM power off, suspend and resume)
+ * that operate for example in fastcall SMC entries of the PSCI
+ * services where non-secure world is not operational. In these
+ * cases we cannot take a mutex and will expect the mutex is
+ * unlocked.
```

Comment:
Maybe should I simplify this description? (and ditto for the equiv inline comment in _regulator.c_)?
```
/*
 * Clock operation may occur at runtime and during specific
 * system power transition (PM power off, suspend and resume)
 * hence use a PM aware mutex.
 */

Reply:
Sure, or even better refer to the description of the `struct mutex_pm_aware`.

Reply:
I've simplified the description.
clk.c: `/* Global clock tree access protection complying the power state transitions */`
regualtor.c: `/* Access protection mutex complying the power state transitions context */`